presto-spi/src/main/java/io/prestosql/spi/predicate/SortedRangeSet.java

```diff
@@ -424,12 +424,15 @@ SortedRangeSet build()
 
             Range current = null;
             for (Range next : ranges) {
+                if (next.isEmpty()) {
```

Comment:
Why do we need this? It should be handled naturally by the call to `span` below.

Reply:
You're right, we don't need it for correctness.
But after this change, some tests result in having an empty range in them, e.g.:
https://github.com/prestosql/presto/blob/df37dfec300192b6f33116a54d77ed89f91013c3/presto-spi/src/test/java/io/prestosql/spi/predicate/TestSortedRangeSet.java#L62-L71
Without skipping empty ranges, the resulting `rangeSet` will be equal to `[(1, 2)]` - which is correct, but IMHO can be optimized into `SortedRangeSet.none()`.
WDYT?

Reply:
P.S.
We can split "empty ranges' handling" into a separate PR, if preferable.

Reply:
Got it. Let's split it up into a separate commit within this PR to keep the logical changes separate.

Reply:
Sounds good - I have split the PR in to two commits:
```
1f57c13d00  Allow merging consecutive 'long' values in SortedRangeSet
c429fdd5d2  Skip empty ranges while constructing SortedRangeSet
```